pynxtools_xps/phi/spe_pro_phi.py

```diff
@@ -0,0 +1,1549 @@
+
+            # Write raw intensities to 'detector'.
+            detector_data_key_child = f"cycles/Cycle_0/scans/Scan_{scan_no}"
+            detector_data_key = (
+                f'{path_map["detector"]}/{detector_data_key_child}/counts'
+            )
+            self._xps_dict[detector_data_key] = intensity
+
+
+class PhiParser:  # pylint: disable=too-few-public-methods
+    """
+    A parser for reading in PHI VersaProbe 4 data in the .spe or
+    .pro format.
+    Tested with Software version SS 3.3.3.2.
+    """
+
+    def __init__(self):
+        """
+        Construct the parser.
+
+        """
+        self.raw_data: str = ""
+        self.spectra: List[Dict[str, Any]] = []
+
+        self.metadata = PhiMetadata()
+
+        self.binary_header_length = 4
+        self.spectra_header_length = 24
+        self.encoding: List[str, int] = ["<f", 4]
+
+        self.binary_header: np.ndarray = None
+        self.spectra_header: np.ndarray = None
+
+    def parse_file(self, file, **kwargs):
+        """
+        Parse the .xy file into a list of dictionaries.
+
+        Parsed data is stored in the attribute 'self.data'.
+        Each dictionary in the data list is a grouping of related
+        attributes. The dictionaries are later re-structured into a
+        nested dictionary that more closely resembles the domain logic.
+
+        Parameters
+        ----------
+        file : str
+            XPS data filepath.
+
+        **kwargs : dict
+            commentprefix : str
+                Prefix for comments in xy file. The default is "#".
+            n_headerlines: int
+                number of header_lines in each data block.
+
+        Returns
+        -------
+        list
+            Flat list of dictionaries containing one spectrum each.
+
+        """
+        self.raw_data = self._read_lines(file)
+        header, data = self._separate_header_and_data()
+        # l = b'\x01'.join(data)
+
+        self.parse_header_into_metadata(header)
+        regions = self.parse_spectral_regions(header)
+        areas = self.parse_spatial_areas(header)
+
+        self.add_regions_and_areas_to_spectra(regions, areas)
+
+        self.parse_binary_header(data)
+        self.parse_data_into_spectra(data)
+
+        self.add_metadata_to_each_spectrum()
+
+        return self.spectra
+
+    def _read_lines(self, file: Union[str, Path]):
+        """
+        Read in all lines from the file as a list of strings.
+
+        Parameters
+        ----------
+        file : str
+            XPS data filepath.
+
+        Returns
+        -------
+        None.
+
+        """
+        with open(file, mode="rb") as txt_file:  # "utf-8"
+            data = txt_file.read()
+
+        return data
+
+    def _separate_header_and_data(self):
+        """
+        Separate header (with metadata) from data.
+
+        Returns
+        -------
+        None.
+
+        """
+        try:  # Windows
+            header, data = self.raw_data.split(b"EOFH\r\n")
```

Comment:
I'm not entirely sure but I think there is a function in python with which you can determine whether the file was saved in windows or unix encoding line endings.

Reply:
I think it is only possible with [chardet](https://pypi.org/project/chardet/). So, in order to keep the dependencies small, I would keep it as it is.